### DISCUSSION TITLE
Guide to create own layout

### DISCUSSION THREAD

**Discussion by dmitriy-8gears** [AnySoftKeyboard/AnySoftKeyboard#3296] [Q&A]
Do you have any guide/tutorial how to create keyboard layout? I didn't find nothing....

**Comment by RusneroV** [comment]
This is all I found [contributing](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/addons/CONTRIBUTING.md) and [key codes](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/api/src/main/java/com/anysoftkeyboard/api/KeyCodes.java).
